### PR TITLE
Update deferred calls to use Callables

### PR DESCRIPTION
**Pull request by KoBeWi**
Somewhat follow-up to #67730

This PR changes `call_deferred()` calls to use Callables, removing many unnecessary method binds as a result. I also changed some usages of MessageQueue to use Callables too.

There is still some usage of old `call_deferred()` left, namely when a method is vararg or private. I also left some MessageQueue calls in SceneTree, as I'm not sure about performance requirements in there.

EDIT:
btw this might fix  some bugs, as there were some methods called that weren't bound, resulting in silent fail.

### CODE REVIEW

**Comment by YuriSizov** [comment]
Ah brilliant, I wanted to do this for so long :) You have an extra empty line, it seems, which fails static checks.

File: scene/gui/spin_box.cpp
Comment (KoBeWi) [inline scene/gui/spin_box.cpp:284]: This is already called automatically when `NOTIFICATION_THEME_CHANGED` is received, the only difference is that here it's deferred for both nodes. I wonder if it could be removed.

File: scene/gui/text_edit.cpp
Comment (AThousandShips) [inline scene/gui/text_edit.cpp:6294]: This seems used in the converter, unsure if we should respect it as it's internal (even marked as such)

File: scene/gui/text_edit.cpp
Comment (AThousandShips) [inline scene/gui/text_edit.cpp:431]: Since this used this syntax should it be `MessageQueue::get_singleton()->push_callable(callable_mp(this, &TextEdit::_emit_caret_changed));`?

File: scene/gui/text_edit.cpp
Comment (KoBeWi) [reply scene/gui/text_edit.cpp:6294]: This method is technically not exposed, so it shouldn't even be in the converter.

File: scene/gui/text_edit.cpp
Comment (KoBeWi) [reply scene/gui/text_edit.cpp:431]: No, I changed MessageQueue calls to Callables to.

File: scene/gui/text_edit.cpp
Comment (AThousandShips) [reply scene/gui/text_edit.cpp:6294]: Strange, see:
```
ERROR: Missing GDScript function in pair (_update_wrap_at - ===> _update_wrap_at_column <===)
```

File: scene/gui/text_edit.cpp
Comment (AThousandShips) [reply scene/gui/text_edit.cpp:431]: Unit tests fail for this class worth checking if using the message queue for these solves that

File: scene/gui/text_edit.cpp
Comment (AThousandShips) [reply scene/gui/text_edit.cpp:6294]: https://github.com/godotengine/godot/blob/2d0ee20ff30461b6b10f6fdfba87511a0ebc6642/editor/renames_map_3_to_4.cpp#L217

File: scene/gui/text_edit.cpp
Comment (YuriSizov) [reply scene/gui/text_edit.cpp:431]: If needed, MessageQueue can still be used with callables, `MessageQueue::get_singleton()->push_callable` is a thing :)

Comment (AThousandShips) [reply scene/gui/text_edit.cpp:431]: Did not resolve this, unsure what got broken here

File: scene/gui/text_edit.cpp
Comment (KoBeWi) [reply scene/gui/text_edit.cpp:6294]: I mean it was a wrong rename. I removed it.

File: scene/gui/text_edit.cpp
Comment (KoBeWi) [reply scene/gui/text_edit.cpp:431]: I need to double-check all removed binds. `callable_mp()` does not support default arguments 😓

File: core/variant/callable.cpp
Comment (KoBeWi) [inline core/variant/callable.cpp:40]: This makes the deferred calls print error when they fail, instead of doing so silently.

Using `callable_mp()` can be very error prone, so this helps finding wrong calls. I just wonder if it has unforeseen side effects.

**Comment by kitbdev** [comment]
I think we can remove the `#include "core/object/message_queue.h"` in text_edit.cpp and other files where MessageQueue was removed.

File: editor/code_editor.cpp
Comment (kitbdev) [inline editor/code_editor.cpp:1575]: When using the Script editor's member overview to go to a method, it doesn't work and I get `core\object\message_queue.cpp:223 - Error calling deferred method: 'CodeEdit::TextEdit::set_caret_line': Method expected 5 arguments, but called with 1.`
Do all the arguments need to be explicitly passed now?
In that case this should be ...`call_deferred(p_line, true, true, 0, 0);`.

Comment (AThousandShips) [reply editor/code_editor.cpp:1575]: It always has to, see above, this is a missed case of default arguments

Comment (AThousandShips) [reply editor/code_editor.cpp:1575]:
```suggestion
	callable_mp((TextEdit *)text_editor, &TextEdit::set_caret_line).call_deferred(p_line, true, true, 0, 0);
```

File: editor/code_editor.cpp
Comment (AThousandShips) [inline editor/code_editor.cpp:1582]:
```suggestion
	callable_mp((TextEdit *)text_editor, &TextEdit::set_caret_column).call_deferred(p_begin);
```
```suggestion
	callable_mp((TextEdit *)text_editor, &TextEdit::set_caret_column).call_deferred(p_begin, true, 0);
```

File: editor/code_editor.cpp
Comment (AThousandShips) [inline editor/code_editor.cpp:520]:
```suggestion
		callable_mp((Control *)replace_text, &Control::grab_focus).call_deferred();
```
Nullopt otherwise

File: editor/code_editor.cpp
Comment (AThousandShips) [inline editor/code_editor.cpp:523]:
```suggestion
		callable_mp((Control *)search_text, &Control::grab_focus).call_deferred();
```

**Comment by KoBeWi** [comment]
I triple-checked everything, but still missed some cases. Thanks for looking into it.

**Review by akien-mga — APPROVED** [review]
Didn't review in depth, but makes sense to me overall. Could be good to merge after 4.3-dev2 is released, giving us some time to spot potential regressions before dev3.

> btw this might fix some bugs, as there were some methods called that weren't bound, resulting in silent fail.

This could actually introduce bugs, if the silent failures are now code that's not supposed to be working, and it working would introduce unwanted changes. So these might be worth flagging for relevant maintainers to look at more closely.

**Comment by KoBeWi** [comment]
> This could actually introduce bugs, if the silent failures are now code that's not supposed to be working, and it working would introduce unwanted changes. So these might be worth flagging for relevant maintainers to look at more closely.

Well, one was in RichTextLabel, I don't remember the other one.

**Comment by akien-mga** [comment]
> > This could actually introduce bugs, if the silent failures are now code that's not supposed to be working, and it working would introduce unwanted changes. So these might be worth flagging for relevant maintainers to look at more closely.
> 
> Well, one was in RichTextLabel, I don't remember the other one.

I see one with `_emit_navigation_debug_changed_signal` in NavigationServer3D, CC @smix8.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**
